src/core/api/index.js

```diff
@@ -62,27 +63,44 @@ export function callApi({
 
   return fetch(apiURL, options)
     .then((response) => {
-      return response.clone().json()
-        .then(
-          (jsonResponse) => ({ response, jsonResponse }),
-          (error) => {
+      function makeTextResponseJson(textResponse) {
+        return { jsonResponse: { text: textResponse }, response };
+      }
+
+      const contentType = response.headers.get('Content-Type').toLowerCase();
+
+      // This is a bit paranoid, but we ensure the API returns a JSON response
+      // (see https://github.com/mozilla/addons-frontend/issues/1701).
+      // If not we'll store the text response in JSON and log an error.
+      // If the JSON parsing fails; we log the error and return an "unknown
+      // error".
+      if (contentType === 'application/json') {
+        return response.json()
+          .then((jsonResponse) => {
+            return { response, jsonResponse };
+          })
+          .catch((error) => {
             log.warn('Could not parse response as JSON:', error);
-            return response.clone().text().then((textResponse) =>
-              ({ response, jsonResponse: { text: textResponse } })
-            );
-          }
-        );
+
+            return makeTextResponseJson(
+              `Unknown Error: Could not parse response from ${apiURL} as JSON`);
```

Comment:
I don't think this is the right thing to do because it relies on the response status to produce an error. If the response body triggers a JSON parse error but its status is 200 then `callApi()` will happily return it as if everything is rosy. I think you should just let this throw an error. If it becomes a problem in real life at least we'll see an error in the console.